csharp/src/Ice/TcpEndpoint.cs

```diff
@@ -184,33 +167,38 @@ internal TcpEndpoint GetEndpoint(TcpAcceptor acceptor)
             }
         }
 
-        private protected override IConnector CreateConnector(EndPoint addr, INetworkProxy? proxy) =>
-            new TcpConnector(Instance, addr, proxy, SourceAddress, Timeout, ConnectionId);
+        internal TcpEndpoint GetEndpoint(TcpAcceptor acceptor)
+        {
+            int port = acceptor.EffectivePort();
+            if (port == Port)
+            {
+                return this;
+            }
+            else
+            {
+                return (TcpEndpoint)CreateEndpoint(Host, port, ConnectionId, HasCompressionFlag, Timeout);
+            }
+        }
```

Comment:
Should this be a `NewPort` method instead that just take the port number?

Reply:
Fixed